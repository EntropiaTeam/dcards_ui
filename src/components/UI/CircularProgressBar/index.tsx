import React, { FC } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Box, CircularProgress } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  general: {
    position: 'absolute',
    display: 'flex',
    zIndex: 1000
  },
  container: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%'
  },
  bottom: {
    position: 'absolute',
    color: theme.palette.grey[theme.palette.type === 'light' ? 200 : 700]
  },
  top: {
    color: 'rgb(200 17 46)',
    animationDuration: '550ms',
    position: 'relative'
  }
}));

const CircularProgressBar: FC = () => {
  const classes = useStyles();

  return (
    <Box className={classes.container}>
      <CircularProgress
        variant="determinate"
        className={classes.bottom}
        size={40}
        thickness={4}
        value={100}
      />
      <CircularProgress
        variant="indeterminate"
        disableShrink
        className={classes.top}
        size={40}
        thickness={4}
      />
    </Box>
  );
};

export default CircularProgressBar;
