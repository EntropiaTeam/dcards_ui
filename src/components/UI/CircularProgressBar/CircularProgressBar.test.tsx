import React from 'react';
import renderer from 'react-test-renderer';
import CircularProgressBar from '.';

describe('<CircularProgressBar />', () => {
  const CircularProgressBarComponent = renderer.create(<CircularProgressBar />).toJSON();

  it('should render CircularProgressBar component correctly', () => {
    expect(CircularProgressBarComponent).toMatchSnapshot();
  });
});
