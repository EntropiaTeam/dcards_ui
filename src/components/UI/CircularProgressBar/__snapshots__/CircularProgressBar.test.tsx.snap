// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularProgressBar /> should render CircularProgressBar component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-container-2"
>
  <div
    aria-valuenow={100}
    className="MuiCircularProgress-root makeStyles-bottom-3 MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(-90deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={
          Object {
            "strokeDasharray": "125.664",
            "strokeDashoffset": "0.000px",
          }
        }
      />
    </svg>
  </div>
  <div
    className="MuiCircularProgress-root makeStyles-top-4 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
